sync/rpc/server.go

```diff
@@ -282,7 +326,23 @@ func (*SyncAgentServer) BackupRemove(ctx context.Context, req *BackupRemoveReque
 	return &Empty{}, nil
 }
 
-func (*SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreRequest) (*Empty, error) {
+func (s *SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreRequest) (*Empty, error) {
+	if strings.HasSuffix(req.SnapshotFileName, ".snap_tmp") {
+		if err := s.PrepareRestore(""); err != nil {
+			logrus.Errorf("failed to preprare backup restore for incremental full restore: %v", err)
+			return nil, err
+		}
+	}
+	finished := false
```

Comment:
We can use `err` for `defer func()` if we add it to the function's signature as a return variable.

Reply:
I quite didn't understand this. Could you please elaborate?

Reply:
it looks like `finished == (err == nil)`, so we should be able to use something like this https://github.com/longhorn/longhorn-manager/blob/master/controller/volume_controller.go#L206